src/main/java/ui/MenuControl.java

```diff
@@ -161,9 +140,41 @@ private void onBoardSaveAs() {
             String boardName = response.get().trim();
             prefs.addBoard(boardName, panelList);
             prefs.setLastOpenBoard(boardName);
+            prefs.setLastOpenBoardPanelInfo(panelList);
             ui.triggerEvent(new BoardSavedEvent());
             logger.info("New board " + boardName + " saved");
             ui.updateTitle();
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Prompt use for whether to save the current board panel info or not
+     * when user is closing current board (i.e. switching to another board)
```

Comment:
According to our convention, should start with `Prompts` since it's a abbreviation for `This method prompts...`